pkg/speccheck/builtin/process_service_instance.go

```diff
@@ -1,17 +1,39 @@
 package builtin
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/atlassian/smith"
 	"github.com/atlassian/smith/pkg/speccheck"
 	"github.com/atlassian/smith/pkg/util"
 	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	SecretParametersChecksumAnnotation = smith.Domain + "/secretParametersChecksum"
+	Disabled                           = "disabled"
 )
 
 type serviceInstance struct {
```

Comment:
All the static functions here and for Deployment are now methods to avoid polluting package's name space. Re-useable ones for hashing are in the `speccheck` package, everything else is not re-useable anyway.